### PR TITLE
(T): add fuzzy test for typing chars in empty file

### PR DESCRIPTION
**Pull request by mkaput**

### CODE REVIEW

File: src/test/kotlin/org/rust/ide/typing/RustTypingEmptyFileFuzzyTest.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package org.rust.ide.typing
+
+import com.intellij.openapi.util.text.StringUtil
+
+class RustTypingEmptyFileFuzzyTest : RustTypingTestCaseBase() {
+    override val dataPath: String = ""
+
+    fun testFuzzy() {
+        // These chars cause problems or require special treating
+        val blacklist = setOf('\b', '\t', '\r', '"', '(', '[', '{')
```
Comment (matklad) [inline src/test/kotlin/org/rust/ide/typing/RustTypingEmptyFileFuzzyTest.kt]: Perhaps we can type backspace more than one to handle these?

**Comment by matklad** [comment]
Splendid! Have you check that this test indeed catches that bug about typing `#`?

**Comment by mkaput** [comment]
Of course, it even maintains stack trace !

File: src/test/kotlin/org/rust/ide/typing/RustTypingEmptyFileFuzzyTest.kt (hunk 1 above)
Comment (mkaput) [inline src/test/kotlin/org/rust/ide/typing/RustTypingEmptyFileFuzzyTest.kt]: It's still special treating, you shouldn't expect `\b` character in file

Comment (mkaput) [inline src/test/kotlin/org/rust/ide/typing/RustTypingEmptyFileFuzzyTest.kt]: `\b` and `\r` are rather specific cases
`\t`'s behaviour depends on user configuration
and the rest are chars we handle with our own typing handlers, and are tested in applicable places

**Comment by matklad** [comment]
Thanks! I llove fuzzy tests!
